frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx

```diff
@@ -26,53 +31,35 @@ const MULTI_SELECT_OPTIONS = [
 
 interface ParameterSettingsProps {
   parameter: UiParameter;
-  onChangeName: (parameterId: string, name: string) => void;
-  onChangeDefaultValue: (parameterId: string, value: unknown) => void;
-  onChangeIsMultiSelect: (parameterId: string, isMultiSelect: boolean) => void;
-  onRemoveParameter: (parameterId: string) => void;
+  onChangeName: (name: string) => void;
+  onChangeDefaultValue: (value: unknown) => void;
+  onChangeIsMultiSelect: (isMultiSelect: boolean) => void;
+  onChangeSourceType: (sourceType: ParameterSourceType) => void;
+  onChangeSourceOptions: (sourceOptions: ParameterSourceOptions) => void;
+  onRemoveParameter: () => void;
 }
 
 const ParameterSettings = ({
   parameter,
   onChangeName,
+  onChangeSourceType,
+  onChangeSourceOptions,
   onChangeDefaultValue,
   onChangeIsMultiSelect,
   onRemoveParameter,
 }: ParameterSettingsProps): JSX.Element => {
-  const parameterId = parameter.id;
-
-  const handleNameChange = useCallback(
```

Comment:
I made all child components of `ParameterSidebar` without without `parameterId` to reduce noise. Now `ParameterSidebar` wraps callbacks and adds `parameterId` where needed.